### ISSUE TITLE
`reboot -s` reboots Qemu...

### ISSUE THREAD

**Issue by drawkula** [minoca/os#92]
`reboot -s` reboots Qemu (v2.1, v2.8) instead of shutting it down.

I noticed this long ago and wanted to keep it back until more important bugs have been fixed but the last days I had some strange trouble with my workhorse VM which seemed to be a rotten filesystem...

At that point I thought it might be influenced by not being able to correctly shutdown the VM.

Currently I use `reboot` and try to close Qemu's window when I see Qemu restarting the VM.

**Comment by evangreen** [comment]
Yeah, shutdown is one of those items on the TODO list. What we need to do to get that working is:
* Create a new hardware module type: HardwareModuleReboot
* Have the ACPI driver register one of these new hardware module types.
* Move the ACPI reset code in HlpArchResetSystem into the new ACPI reboot module.
* To support shutdown, grab the _S5 package from the ACPI namespace to know the shutdown value.
* Write that shutdown value into one of the fixed registers (I forget which one, PM1a maybe?)

The HardwareModuleReboot won't exclusively be for ACPI, some of the SOCs whose reset control is out over I2C on some PMIC would also use this mechanism. The RK808 comes to mind.

Regarding the rotten file system, that method you're using should work. What I do is make sure the first number next to `Cache: ` on the top line of the screen is zero. That represents the amount of "dirty" page cache data. If that's zero, you can pretty much flip the machine off forcefully. Probably things got hairy when it got turned off with that number being non-zero.

**Comment by evangreen** [comment]
The latest nightlies have PC shutdown support. Feel free to give them a try, and reopen this issue or file a new one if something isn't working.
